### PR TITLE
User Story 2

### LINKED ISSUE #5: User Story 2, Visitor Navigation
As a visitor
I see a navigation bar
This navigation bar includes links for the following:
- a link to return to the welcome / home page of the application ("/")
- a link to browse all items for sale ("/items")
- a link to see all merchants ("/merchants")
- a link to my shopping cart ("/cart")
- a link to log in ("/login")
- a link to the user registration page ("/register")

Next to the shopping cart link I see a count of the items in my cart

### PR DESCRIPTION
**Pull request by brian-greeson**
- Added logic to navbar in application layout view to show correct links to visitors
- Updated application controller with `current_user` on helper methods
- Closes: #5

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  helper_method :cart
+  helper_method :cart, :current_user
```
Comment (brian-greeson) [inline app/controllers/application_controller.rb:4]: We are going to need access to this in our views.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,10 +10,12 @@
 
   <body>
     <nav class = "topnav">
+      <%= link_to "Home", "/" %>
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
       <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register", "/register" %>
+      <%= link_to "Register", register_path if !current_user %>
+      <%= link_to "Login", login_path if !current_user %>
```
Comment (brian-greeson) [inline app/views/layouts/application.html.erb:18]: This change will make more sense in the next user story

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (kelshadarby) [reply app/controllers/application_controller.rb:4]: This is a great catch!

Comment (fredrondina) [reply app/controllers/application_controller.rb:4]: nice change

**Review by kelshadarby — APPROVED** [review]

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (fredrondina) [reply app/views/layouts/application.html.erb:18]: good work getting these buttons in, the log out button will be added in US 16

**Review by fredrondina — APPROVED** [review]
nice work, the application view looks good

**Pull request merged**